features/mask/core/src/main/java/org/apache/shardingsphere/mask/algorithm/cover/TelephoneRandomReplaceAlgorithm.java

```diff
@@ -0,0 +1,92 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.shardingsphere.mask.algorithm.cover;
+
+import com.google.common.base.Splitter;
+import com.google.common.base.Strings;
+import lombok.Getter;
+import org.apache.shardingsphere.mask.algorithm.MaskAlgorithmUtil;
+import org.apache.shardingsphere.mask.exception.algorithm.MaskAlgorithmInitializationException;
+import org.apache.shardingsphere.mask.spi.MaskAlgorithm;
+
+import java.util.Comparator;
+import java.util.List;
+import java.util.Properties;
+import java.util.Random;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * Telephone random replace algorithm.
+ */
+public final class TelephoneRandomReplaceAlgorithm implements MaskAlgorithm<Object, String> {
+    
+    private static final String NETWORK_NUMBER = "network-number";
+    
+    private Set<String> networkNumbers;
+    
+    private List<Integer> networkNumberLength;
+    
+    @Getter
+    private Properties props;
+    
+    @Override
+    public void init(final Properties props) {
+        this.props = props;
+        createNetworkNumbers(props);
+    }
+    
+    private void createNetworkNumbers(final Properties props) {
+        MaskAlgorithmUtil.checkAtLeastOneCharConfig(props, NETWORK_NUMBER, getType());
+        networkNumbers = Splitter.on(",").trimResults().splitToStream(props.getProperty(NETWORK_NUMBER)).map(networkNumber -> isNumeric(networkNumber)).collect(Collectors.toSet());
+        networkNumberLength = networkNumbers.stream().map(networkNumber -> String.valueOf(networkNumber).length()).distinct().sorted(Comparator.reverseOrder()).collect(Collectors.toList());
+    }
+    
+    private String isNumeric(final String networkNumber) {
+        try {
+            Integer.parseInt(networkNumber);
+            return networkNumber;
+        } catch (NumberFormatException e) {
+            throw new MaskAlgorithmInitializationException(getType(), "network-number can only be numbers and ,");
+        }
+    }
+    
+    @Override
+    public String mask(final Object plainValue) {
+        String result = null == plainValue ? null : String.valueOf(plainValue);
+        if (Strings.isNullOrEmpty(result)) {
+            return result;
+        }
+        Random random = new Random();
+        char[] chars = result.toCharArray();
+        for (Integer length : networkNumberLength) {
+            if (networkNumbers.contains(result.substring(0, length))) {
+                for (int i = length; i != -1 && i < chars.length; i++) {
+                    chars[i] = Character.forDigit(random.nextInt(10), 10);
+                }
+                continue;
+            }
+        }
+        return new String(chars);
+    }
+    
+    @Override
+    public String getType() {
+        return "TELEPHONE_RANDOM_REPLACE";
```

Comment:
Hi @gxxiong, can we add this new algorithm to mask built-in algorithm document and dev-manual document?

- https://shardingsphere.apache.org/document/current/cn/user-manual/common-config/builtin-algorithm/mask/
- https://shardingsphere.apache.org/document/current/cn/dev-manual/mask/